### ISSUE TITLE
PHP Notice's

### ISSUE THREAD

**Issue by DreamInk-Offical** [soflyy/oxygen-bugs-and-features#1719]
Hey! I'm getting some PHP Notices in my debug log.
Perhaps someone else already mentioned these:

<pre><code>[17-Nov-2020 14:26:09 UTC] PHP Notice:  Undefined variable: show_gradient in \oxygen\component-framework\toolbar\views\background.view.php on line 167
[17-Nov-2020 14:26:24 UTC] PHP Notice:  ob_end_flush(): failed to send buffer of zlib output compression (2) in wp-includes\functions.php on line 4673
[17-Nov-2020 14:26:26 UTC] PHP Notice:  Undefined index: html_template in \oxygen\component-framework\api\oxygen.element.class.php on line 295
[17-Nov-2020 14:26:26 UTC] PHP Notice:  Undefined index: inlineJS in \oxygen\component-framework\api\oxygen.element.class.php on line 316
[17-Nov-2020 14:26:26 UTC] PHP Notice:  Undefined index: inlineJS in \oxygen\component-framework\api\oxygen.element.class.php on line 317
[17-Nov-2020 14:26:26 UTC] PHP Notice:  Undefined property: OxygenElement::$registeredSelectors in \oxygen\component-framework\api\oxygen.element.class.php on line 348
[17-Nov-2020 14:26:31 UTC] PHP Notice:  Indirect modification of overloaded element of WP_Hook has no effect in \oxygen\component-framework\components\classes\code-block.class.php on line 34
[17-Nov-2020 14:26:31 UTC] PHP Notice:  Undefined index: queryOptions in \oxygen\component-framework\api\single.template.php on line 20
[17-Nov-2020 14:26:31 UTC] PHP Notice:  Undefined index: children in \oxygen\component-framework\api\oxygen.element.class.php on line 874
[17-Nov-2020 14:26:32 UTC] PHP Notice:  Undefined index: queryOptions in \oxygen\component-framework\components\layouts\shortcode.php on line 51
[17-Nov-2020 14:26:32 UTC] PHP Notice:  Undefined variable: ct_template_id in \oxygen\component-framework\api\oxygen.element.class.php on line 918
[17-Nov-2020 14:26:32 UTC] PHP Notice:  Undefined variable: ct_template_id in \oxygen\component-framework\api\oxygen.element.class.php on line 919
</code></pre>

These errors occur when I'm loading the Oxygen Editor.

**Comment by swinggraphics** [comment]
A lot of notices appear on the frontend and backend with the latest version, too. Mostly "undefined index" for line-height, gradient colors, or other CSS properties. Makes debugging my own work much, much harder.

**Comment by zakalwe2016** [comment]
Just want to second this. This doesn't have many votes for fixing but clean logs are essential for development. I haven't looked into the code but I would imagine these are just simple checks. Sadly we can't provide PRs for this. Fixing these should also aid the development of Oxygen itself. I'm not sure how the devs are handling this but it must be frustrating for them to have to wade through unecessary notices.

**Comment by wpsumo** [comment]
@KittenCodes @Spellhammer if you fire up query monitor in the builder UI, I see the following with Oxygen 3.7.1 and was not reported or resolved in 3.8 as far as I know?


LevelAllWarningNoticeStrictDeprecated | Message | Count | Location | ComponentAllPlugin: oxygen
-- | -- | -- | -- | --
Notice | Undefined variable: tag | 1 | +wp-content/plugins/oxygen/component-framework/toolbar/views/position/position.flex-layout.view.php:11 | Plugin: oxygen
Notice | Undefined variable: tag | 1 | +wp-content/plugins/oxygen/component-framework/toolbar/views/position/position.flex-layout.view.php:31 | Plugin: oxygen
Notice | Undefined variable: tag | 1 | +wp-content/plugins/oxygen/component-framework/toolbar/views/position/position.flex-layout.view.php:53 | Plugin: oxygen
Notice | Undefined variable: tag | 1 | +wp-content/plugins/oxygen/component-framework/toolbar/views/position/position.flex-layout.view.php:11 | Plugin: oxygen
Notice | Undefined variable: tag | 1 | +wp-content/plugins/oxygen/component-framework/toolbar/views/position/position.flex-layout.view.php:31 | Plugin: oxygen
Notice | Undefined variable: tag | 1 | +wp-content/plugins/oxygen/component-framework/toolbar/views/position/position.flex-layout.view.php:53 | Plugin: oxygen
Notice | Undefined variable: tag | 1 | +wp-content/plugins/oxygen/component-framework/toolbar/views/position/position.flex-layout.view.php:11 | Plugin: oxygen
Notice | Undefined variable: tag | 1 | +wp-content/plugins/oxygen/component-framework/toolbar/views/position/position.flex-layout.view.php:31 | Plugin: oxygen
Notice | Undefined variable: tag | 1 | +wp-content/plugins/oxygen/component-framework/toolbar/views/position/position.flex-layout.view.php:53 | Plugin: oxygen
Notice | Undefined variable: show_gradient | 1 | +wp-content/plugins/oxygen/component-framework/toolbar/views/background.view.php:167 | Plugin: oxygen

**Comment by controlauk** [comment]
im getting in 3.8

PHP Notice: Indirect modification of overloaded element of WP_Hook has no effect in /home/stagingcontrola/public_html/wp-content/plugins/oxygen/component-framework/components/classes/code-block.class.php on line 34

**Comment by Spellhammer** [comment]
> im getting in 3.8
> 
> PHP Notice: Indirect modification of overloaded element of WP_Hook has no effect in /home/stagingcontrola/public_html/wp-content/plugins/oxygen/component-framework/components/classes/code-block.class.php on line 34

This is almost certainly the result of some code you've written in a Code Block, and not a core problem with Oxygen. We can help you with that at https://oxygenbuilder.com/support if you'd like.

**Comment by Spellhammer** [comment]
Fixed in Oxygen 4.1 beta 1.

**Comment by BertMammoet** [comment]
I am still getting a notice:
`[18-Aug-2023 15:08:16 UTC] PHP Notice:  ob_end_flush(): failed to send buffer of zlib output compression (1) in C:\*******\*********\****\wp-includes\functions.php on line 5349
[18-Aug-2023 15:08:16 UTC] PHP Stack trace:
[18-Aug-2023 15:08:16 UTC] PHP   1. {main}() C:\*******\*********\****\wp-admin\admin-ajax.php:0
[18-Aug-2023 15:08:16 UTC] PHP   2. do_action() C:\*******\*********\****\wp-admin\admin-ajax.php:188
[18-Aug-2023 15:08:16 UTC] PHP   3. WP_Hook->do_action() C:\*******\*********\****\wp-includes\plugin.php:517
[18-Aug-2023 15:08:16 UTC] PHP   4. WP_Hook->apply_filters() C:\*******\*********\****\wp-includes\class-wp-hook.php:334
[18-Aug-2023 15:08:16 UTC] PHP   5. ct_get_svg_icon_sets() C:\*******\*********\****\wp-includes\class-wp-hook.php:310
[18-Aug-2023 15:08:16 UTC] PHP   6. shutdown_action_hook() C:\*******\*********\****\wp-includes\load.php:0
[18-Aug-2023 15:08:16 UTC] PHP   7. do_action() C:\*******\*********\****\wp-includes\load.php:1252
[18-Aug-2023 15:08:16 UTC] PHP   8. WP_Hook->do_action() C:\*******\*********\****\wp-includes\plugin.php:517
[18-Aug-2023 15:08:16 UTC] PHP   9. WP_Hook->apply_filters() C:\*******\*********\****\wp-includes\class-wp-hook.php:334
[18-Aug-2023 15:08:16 UTC] PHP  10. wp_ob_end_flush_all() C:\*******\*********\****\wp-includes\class-wp-hook.php:310
[18-Aug-2023 15:08:16 UTC] PHP  11. ob_end_flush() C:\*******\*********\****\wp-includes\functions.php:5349
[18-Aug-2023 15:08:16 UTC] PHP Notice:  ob_end_flush(): failed to send buffer of zlib output compression (1) in C:\*******\*********\****\wp-includes\functions.php on line 5349
[18-Aug-2023 15:08:16 UTC] PHP Stack trace:
[18-Aug-2023 15:08:16 UTC] PHP   1. {main}() C:\*******\*********\****\wp-admin\admin-ajax.php:0
[18-Aug-2023 15:08:16 UTC] PHP   2. do_action() C:\*******\*********\****\wp-admin\admin-ajax.php:188
[18-Aug-2023 15:08:16 UTC] PHP   3. WP_Hook->do_action() C:\*******\*********\****\wp-includes\plugin.php:517
[18-Aug-2023 15:08:16 UTC] PHP   4. WP_Hook->apply_filters() C:\*******\*********\****\wp-includes\class-wp-hook.php:334
[18-Aug-2023 15:08:16 UTC] PHP   5. ct_get_svg_icon_sets() C:\*******\*********\****\wp-includes\class-wp-hook.php:310
[18-Aug-2023 15:08:16 UTC] PHP   6. shutdown_action_hook() C:\*******\*********\****\wp-includes\load.php:0
[18-Aug-2023 15:08:16 UTC] PHP   7. do_action() C:\*******\*********\****\wp-includes\load.php:1252
[18-Aug-2023 15:08:16 UTC] PHP   8. WP_Hook->do_action() C:\*******\*********\****\wp-includes\plugin.php:517
[18-Aug-2023 15:08:16 UTC] PHP   9. WP_Hook->apply_filters() C:\*******\*********\****\wp-includes\class-wp-hook.php:334
[18-Aug-2023 15:08:16 UTC] PHP  10. wp_ob_end_flush_all() C:\*******\*********\****\wp-includes\class-wp-hook.php:310
[18-Aug-2023 15:08:16 UTC] PHP  11. ob_end_flush() C:\*******\*********\****\wp-includes\functions.php:5349
`

I was able to fix this to add to:
**wp-content\plugins\oxygen\component-framework\includes\ajax.php @1445**
I have added:
`remove_action( 'shutdown', 'wp_ob_end_flush_all', 1 );`
to be called before:
`add_action('wp_ajax_ct_get_svg_icon_sets', 'ct_get_svg_icon_sets');`

Now the notices are gone, could you resolve this is a future update?
